neverbleed.c

```diff
@@ -505,10 +527,28 @@ void neverbleed_transaction_write(neverbleed_t *nb, neverbleed_iobuf_t *buf)
     iobuf_transaction_write(buf, thdata);
 }
 
+static int exdata_idx = -1;
```

Comment:
```suggestion
static volatile int exdata_idx = -1;
```

We need volatile, otherwise there is no guarantee that the value would be read after being written to.  `pthread_mutex_unlock` is an external function, but the compiler can avoid re-reading `exdata_idx` across the invocation of that function because it can learn by compiling neverbleed.c that `exdata_idx` is never taken its address and therefore that functions not defined in this source file is never going to touch the value.